Services/Dashboard/classes/class.ilDashboardGUI.php

```diff
@@ -365,109 +330,101 @@ public function getRightColumnHTML(): string
         }
         $this->show();
     }
 
     public function jumpToProfile(): void
     {
-        $this->ctrl->redirectByClass("ilpersonalprofilegui");
+        $this->ctrl->redirectByClass(ilPersonalProfileGUI::class);
     }
 
     public function jumpToPortfolio(): void
     {
-        // incoming back link from shared resource
-        $cmd = "";
-        if ($this->requested_dsh != "") {
-            $this->ctrl->setParameterByClass("ilportfoliorepositorygui", "shr_id", $this->requested_dsh);
-            $cmd = "showOther";
+        $cmd = '';
+        if ($this->requested_dsh !== '') {
+            $this->ctrl->setParameterByClass(ilPortfolioRepositoryGUI::class, 'shr_id', $this->requested_dsh);
+            $cmd = 'showOther';
         }
 
-        // used for goto links
         if ($this->requested_prt_id > 0) {
-            $this->ctrl->setParameterByClass("ilobjportfoliogui", "prt_id", $this->requested_prt_id);
-            $this->ctrl->setParameterByClass("ilobjportfoliogui", "gtp", $this->requested_gtp);
-            $this->ctrl->redirectByClass(array("ilportfoliorepositorygui", "ilobjportfoliogui"), "preview");
+            $this->ctrl->setParameterByClass(ilObjPortfolioGUI::class, 'prt_id', $this->requested_prt_id);
+            $this->ctrl->setParameterByClass(ilObjPortfolioGUI::class, 'gtp', $this->requested_gtp);
+            $this->ctrl->redirectByClass([ilPortfolioRepositoryGUI::class, ilObjPortfolioGUI::class], 'preview');
         } else {
-            $this->ctrl->redirectByClass("ilportfoliorepositorygui", $cmd);
+            $this->ctrl->redirectByClass(ilPortfolioRepositoryGUI::class, $cmd);
         }
     }
 
     public function jumpToSettings(): void
     {
-        $this->ctrl->redirectByClass("ilpersonalsettingsgui");
+        $this->ctrl->redirectByClass(ilPersonalSettingsGUI::class);
     }
 
     public function jumpToNews(): void
     {
-        $this->ctrl->redirectByClass("ilpdnewsgui");
+        $this->ctrl->redirectByClass(ilPDNewsGUI::class);
     }
 
     public function jumpToCalendar(): void
     {
         global $DIC;
-        $request = $DIC->http()->request();
-
-        $query_params = $request->getQueryParams();
+        $query_params = $DIC->http()->request()->getQueryParams();
 
-        if (array_key_exists("cal_view", $query_params) && $query_params["cal_view"]) {
-            $cal_view = $query_params["cal_view"];
-            $this->ctrl->setParameter($this, "cal_view", $cal_view);
+        if (array_key_exists('cal_view', $query_params) && $query_params['cal_view']) {
+            $cal_view = $query_params['cal_view'];
+            $this->ctrl->setParameter($this, 'cal_view', $cal_view);
         }
 
-        if (!empty($query_params["cal_agenda_per"])) {
-            $cal_period = $query_params["cal_agenda_per"];
-            $this->ctrl->setParameter($this, "cal_agenda_per", $cal_period);
+        if (!empty($query_params['cal_agenda_per'])) {
```

Comment:
Please change this method to either use empty(...) or `array_key_exists(...) && ...`. Because they are both the same.

Reply:
As far as i know its not the same since empty also checks for "empty" values like `null`, `""` and so on.

Further im pretty sure empty should be faster since its conceptional check is less complex.

Reply:
The first expression is: `array_key_exits($key, $query_params) && $query_params[$key]` so it is equivalent to `empty). I don't care which to choose but it should be the same in both cases.

Reply:
Wich first expression?

Reply:
https://github.com/ILIAS-eLearning/ILIAS/blob/2c335dde69dd84383621f5cc6596e1095d4f673b/Services/Dashboard/classes/class.ilDashboardGUI.php#L398